app/main.py

```diff
@@ -1,2 +1,35 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.hashmap: list = [[] for _ in range(self.capacity)]
+        self.length = 0
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        hash_key = hash(key) % self.capacity
```

Comment:
There is no need to obtain a modulo from the key hash since after resizing all elements would be reordered in the hashmap and it introduces additional load to the elements adding. You should avoid reordering by storing the hash directly.